src/main/java/org/opensearch/knn/index/mapper/CompressionLevel.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.mapper;
+
+import lombok.AllArgsConstructor;
+
+import java.util.Locale;
+
+/**
+ * Enum representing the compression level for float vectors. Compression in this sense refers to compressing a
+ * full precision value into a smaller number of bits. For instance. "16x" compression would mean that 2 bits would
+ * need to be used to represent a 32-bit floating point number.
+ */
+@AllArgsConstructor
+public enum CompressionLevel {
+    NOT_CONFIGURED(-1),
+    x1(1),
+    x2(2),
+    x4(4),
+    x8(8),
+    x16(16),
+    x32(32);
+
+    /**
+     * Default is set to 1x and is a noop
+     */
+    private static final CompressionLevel DEFAULT = x1;
+
+    /**
+     * Get the compression level from a string representation. The format for the string should be "Nx", where N is
+     * the factor by which compression should take place
+     *
+     * @param name String representation of the compression level
+     * @return CompressionLevel enum value
+     */
+    public static CompressionLevel fromString(String name) {
+        if (name == null) {
+            return NOT_CONFIGURED;
+        }
+        for (CompressionLevel config : CompressionLevel.values()) {
+            if (config.toString() != null && config.toString().equals(name)) {
+                return config;
+            }
+        }
+        throw new IllegalArgumentException(String.format(Locale.ROOT, "Invalid compression level: \"[%s]\"", name));
+    }
+
+    private final int compressionLevel;
+
+    @Override
+    public String toString() {
+        if (this == NOT_CONFIGURED) {
+            return null;
+        }
+        return compressionLevel + "x";
+    }
```

Comment:
we are returning 32x here when enum is x32. Can we have like 1 notation. My recommendation will be to use 32x rather than x32.

Reply:
Enum needs to start with alpha character unfortunately.

Reply:
So I think 32x as string is fine, then x32 is good for name

Reply:
> Enum needs to start with alpha character unfortunately.

Didn't know this. Feels pretty weird.